File: python/datastructure/otterop/datastructure/linked_list_node.py

class LinkedListNode:
                
    def __init__(self, value):
        self._value = value

    def list(self):
        return self._list

    def _set_list(self, list):
        self._list = list

    def prev(self):
        return self._prev

    def _set_prev(self, node):
        self._prev = node

    def next(self):
        return self._next

    def _set_next(self, node):
        self._next = node

    def value(self):
        return self._value

File: java/datastructure/src/main/java/otterop/datastructure/LinkedListNode.java
package otterop.datastructure;

public class LinkedListNode<T> {
        private LinkedList<T> list;
        private LinkedListNode<T> prev;
        private LinkedListNode<T> next;
        private T value;

        public LinkedListNode(T value) {
            this.prev = null;
            this.next = null;
            this.value = value;
        }

        public LinkedList<T> list() {
            return this.list;
        }

        void setList(LinkedList<T> list) {
            this.list = list;
        }

        public LinkedListNode<T> prev() {
            return this.prev;
        }

        void setPrev(LinkedListNode<T> node) {
            this.prev = node;
        }

        public LinkedListNode<T> next() {
            return this.next;
        }

        void setNext(LinkedListNode<T> node) {
            this.next = node;
        }

        public T value() {
            return this.value;
        }
}
